core/base/src/TApplication.cxx

```diff
@@ -628,21 +630,416 @@ TApplication::EExitOnException TApplication::ExitOnException(TApplication::EExit
-   Printf("             which [file] : shows path of macro file");
+static TString URLforEnumerator(TString &className, const TString &enumeratorName, TDataMember *dataMember, EUrl scopeType)
+{
+   // We create a TString with the name of the class and the enumeration from which "enumeratorName" is.
+   TString classEnumeration = dataMember->GetTrueTypeName();
+   TString md5EnumClass;
+   // If we have anonymous enumeration we need other syntax ("className::@1@1").
+   if ((classEnumeration.Contains("(anonymous)"))) {
+      // FIXME: need to invetigate the numbering scheme.
+      md5EnumClass.Append(className);
+      md5EnumClass.Append("::@1@1");
+   } else {
+      // If the enumeration is not anonymous we put "className::Enumeration" in a TString,
+      // which will be crypted with MD5 later.
+      md5EnumClass.Append(classEnumeration);
+      TString delim("::");
+      // We extract the part after "::" (this is the enumerator name).
+      TObjArray enumOnly(*classEnumeration.Tokenize(delim));
+      TString enumOnlyName = (enumOnly.Last())->GetName();
+      // The syntax is "Class::EnumeratorEnumerator
+      md5EnumClass.Append(enumOnlyName);
+   }
+   // The next part of the URL is crypted "@ className::EnumeratorEnumerator".
+   TString md5Enumerator("@ ");
+   md5Enumerator.Append(className);
+   md5Enumerator.Append("::");
+   md5Enumerator.Append(enumeratorName);
+   md5Enumerator.Append(enumeratorName);
+   // We make the URL for the "className".
+   className.ReplaceAll("::","_1_1");
+   TString url = UrlGenerator(className, scopeType);
+   // Then we have to append the crypted text for the enumerator.
+   url.Append("#a");
+   url.Append(md5EnumClass.MD5());
+   // We append "a" and then the next crypted text.
+   url.Append("a");
+   url.Append(md5Enumerator.MD5());
+   return url;
+}
 }
 
+namespace {
 ////////////////////////////////////////////////////////////////////////////////
+/// The function generates URL for enumeration. The hashed text consist of:
+/// "Class::EnumerationEnumeration".
+///
+/// \param[in] className the name of the class
+/// \param[in] enumeration the name of the enumeration
+/// \param[in] scopeType enumerator for class or namespace
+
+
+static TString URLforEnumeration(TString &className, const TString &enumeration, EUrl scopeType){
+   // The URL consists of URL for the "className", "#a" and crypted with MD5 text.
+   // The text is "Class::EnumerationEnumeration.
+   TString md5Enumeration(className);
+   md5Enumeration.Append("::");
+   md5Enumeration.Append(enumeration);
+   md5Enumeration.Append(enumeration);
+   // We make the URL for the class "className".
+   className.ReplaceAll("::","_1_1");
+   TString url(UrlGenerator(className, scopeType));
+   // Then we have to append "#a" and the crypted text.
+   url.Append("#a");
+   url.Append(md5Enumeration.MD5());
+   return url;
+}
+}
+
+namespace {
+enum EClassURL {kURLforMethod, kURLforConstructor};
```

Comment:
`kURLforConstructor` is used for both constructor and destructor; the enumerator name does not reflect that. Consider renaming to `kURLforStructor`.